platform/ios/src/MGLMapView.mm

```diff
@@ -1169,6 +1169,10 @@ - (void)touchesBegan:(__unused NS_SET_OF(UITouch *) *)touches withEvent:(__unuse
 {
     _changeDelimiterSuppressionDepth = 0;
     _mbglMap->setGestureInProgress(false);
+    if (self.userTrackingState == MGLUserTrackingStateBegan)
+    {
+        [self setUserTrackingMode:MGLUserTrackingModeNone animated:NO];
```

Comment:
This code runs for every gesture, unconditionally. However, we intentionally stay in user tracking mode for gestures other than panning. For example, it should be possible to remain in location tracking mode after pinching or rotating and remain in heading tracking mode after tilting.